File: app/src/main/java/com/rockon999/android/firetv/leanbacklauncher/apps/RowType.java
package com.rockon999.android.firetv.leanbacklauncher.apps;

import android.util.SparseArray;

/**
 * Created by rockon999 on 3/7/18.
 */

public enum RowType {

    SEARCH(0),
    NOTIFICATIONS(1),
    PARTNER(2),
    APPS(3),
    GAMES(4),
    SETTINGS(5),
    INPUTS(6),
    FAVORITES(7),
    MUSIC(8),
    VIDEO(9),
    ACTUAL_NOTIFICATIONS(10);

    /**
     * Created by rockon999 on 2/18/18.
     */
    private final int code;

    RowType(int code) {
        this.code = code;
    }

    private static final SparseArray<RowType> intToType = new SparseArray<>();

    public int getCode() {
        return code;
    }

    public static RowType fromRowCode(int code) {
        RowType category = intToType.get(code);
        return category == null ? RowType.APPS : category;
    }

    public static RowType fromName(String name) {
        if (name == null) return null;
        return valueOf(name.toUpperCase().trim());
    }

    static {
        for (RowType type : values()) {
            intToType.put(type.code, type);
        }
    }
}

File: app/src/main/java/com/amazon/tv/firetv/leanbacklauncher/apps/RowType.kt
package com.amazon.tv.firetv.leanbacklauncher.apps

import android.util.SparseArray

enum class RowType(val code: Int) {
    SEARCH(0),
    NOTIFICATIONS(1),
    PARTNER(2),
    APPS(3),
    GAMES(4),
    SETTINGS(5),
    INPUTS(6),
    FAVORITES(7),
    MUSIC(8),
    VIDEO(9),
    ACTUAL_NOTIFICATIONS(10);

    companion object {
        private val intToType = SparseArray<RowType>()

        fun fromRowCode(code: Int): RowType {
            val category = intToType[code]
            return category ?: APPS
        }

        fun fromName(name: String?): RowType? {
            return if (name == null) null else valueOf(name.toUpperCase().trim { it <= ' ' })
        }

        val VALUES = values()
        init {
            for (type in VALUES) {
                intToType.put(type.code, type)
            }
        }
    }
}